magefile.go

```diff
@@ -818,15 +818,20 @@ func packageAgent(platforms []string, packagingFn func()) {
 }
 
 func copyComponentSpecs(componentName, versionedDropPath string) (string, error) {
-	sourceSpecFile := filepath.Join("specs", componentName+specSuffix)
-	targetPath := filepath.Join(versionedDropPath, componentName+specSuffix)
-	err := devtools.Copy(sourceSpecFile, targetPath)
-	if err != nil {
-		return "", errors.Wrapf(err, "failed copying spec file %q to %q", sourceSpecFile, targetPath)
+	specFileName := componentName + specSuffix
+	targetPath := filepath.Join(versionedDropPath, specFileName)
+
+	if _, err := os.Stat(targetPath); err != nil {
+		// spec not present copy from local
+		sourceSpecFile := filepath.Join("specs", specFileName)
+		err := devtools.Copy(sourceSpecFile, targetPath)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed copying spec file %q to %q", sourceSpecFile, targetPath)
+		}
```

Comment:
The spec file in the downloaded component archive takes precedence over the one in the agent repository. Is that understanding correct?

The contents of the two files don't match so I want to update https://github.com/elastic/elastic-agent-shipper/issues/222 to specify what happens now.

Reply:
yes whatever is in the package is kept there, otherwise we copy what we have as a backup. priority is for spec file from product repo maintained by owners

Reply:
removed shipper.yml, it does not even match agent shipper name so it cannot be used at all

Reply:
i will bring it back, tests are using spec file